File: gcj_2017/QR_B.m
// Google Code Jam 2017 Qualification Round
// Problem B. Tidy Numbers

#import <Foundation/Foundation.h>
#import <stdio.h>

@interface Solver : NSObject {

}
+ (NSInteger) solve:(NSInteger)n;
@end

@implementation Solver
+ (NSInteger) solve:(NSInteger)n {
    for (NSInteger d = 1; d <= n; d *= 10) {
        NSInteger a = (n / d) % 10, b = (n / d / 10) % 10;
        if (a < b) {
            NSInteger c = n - (n % (d * 10));
            n = c - 1;
        }
    }
    return n;
}
@end

int main(void) {
    @autoreleasepool {
        char buffer[1024];
        fgets(buffer, 1024, stdin);
        NSInteger T = atoll(buffer);
        for (NSInteger t = 1; t <= T; ++t) {
            fgets(buffer, 1024, stdin);
            NSInteger n = atoll(buffer);
            NSInteger ans = [Solver solve:n];
            printf("Case #%ld: %ld\n", (long)t, (long)ans);
        }
    }
    return 0;
}

File: gcj_2017/QR_B.swift
// Google Code Jam 2017 Qualification Round
// Problem B. Tidy Numbers

func readInt() -> Int {
  return Int(readLine() ?? "") ?? 0
}

func solve(n: Int) -> Int {
  var d = 1, r = n
  while (d <= r) {
    let a = (r / d) % 10, b = (r / d / 10) % 10
    if (a < b) {
      let c = r - (r % (d * 10))
      r = c - 1
    }
    d *= 10
  }
  return r
}

let T = readInt()
for t in 1...T {
  let ans = solve(n: readInt())
  print("Case #\(t): \(ans)")
}
